src/Common/tests/System/PlatformDetection.cs

```diff
@@ -31,8 +31,24 @@ public static partial class PlatformDetection
         public static bool IsNetBSD => RuntimeInformation.IsOSPlatform(OSPlatform.Create("NETBSD"));
         public static bool IsOpenSUSE => IsDistroAndVersion("opensuse");
         public static bool IsUbuntu => IsDistroAndVersion("ubuntu");
-        public static bool IsWindowsNanoServer => (IsWindows && !File.Exists(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "regedit.exe")));
+        public static bool IsWindowsNanoServer => (IsWindows && IsNotWindowsIoTCore && !File.Exists(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "regedit.exe")));
         public static bool IsNotWindowsNanoServer => !IsWindowsNanoServer;
+        public static bool IsWindowsIoTCore
+        {
+            get
+            {
+                int productType;
+                Assert.True(GetProductInfo(Environment.OSVersion.Version.Major, Environment.OSVersion.Version.Minor, 0, 0, out productType));
+                if ((productType == PRODUCT_IOTUAPCOMMERCIAL) ||
+                    (productType == PRODUCT_IOTUAP))
+                {
+                    return true;
+                }
+                return false;
+            }
+        }
+        public static bool IsNotWindowsIoTCore => !IsWindowsIoTCore;
+        public static bool IsDrawingSupported => (IsNotWindowsNanoServer && IsNotWindowsNanoServer);
```

Comment:
@danmosemsft / @paulmon Was this a typo here that `IsNotWindowsNanoServer` is checked twice? Also, according to previous comments, should it also be checking `IsNotWindowsIoTCore`?